### DISCUSSION TITLE
custom registries setup works locally only

### DISCUSSION THREAD

**Discussion by AbdulAbCellera** [microsoft/vcpkg#21629] [Q&A] [answered]
I have the current setup

- custom-pkg-repo:
     - `dd` (a commit that is tagged with a version)
        - added all binaries
     - `62`
          - setting the repo large files settings
     - `9e` 
          - added readme file   

- custom-registry-repo:
     - `e5`
        - versions/c-/custom-pkg.son
        - versions/baseline.json
     - `9f`
          - port/custom-pkg/vcpkg.json
          - port/custom-pkg/portfile.cmake (references the `dd` commit in the `custom-pkg-repo`)
     - `cf` 
          - added readme file
- project-repo:
     - `aa`
        - vcpkg-configuration.json ()
               "default-registry": {
                    "kind": "builtin",
                    "baseline": "some commit"
               },
               "registries": [
                    {
                         "kind": "git",
                         "repository": "https://github.com/<ORGANIZATION>/custom-registry-repo",
                         "baseline": "`e5`",
                         "packages": [ "custom-pkg"]
                    }
               ]
---------------------------------------------------------------------

- I am able to download and build (just copy bins to vcpkg folders) the pkg on my local machine. 
- On master Jenkins, the cmake configuration/build is stuck at this step
    00:05:11.952  -- Running vcpkg install
    00:05:11.952  Fetching baseline information from https://github.com/<ORGANIZATION>/custom-registry-repo...
- On the Jenkins build agents, I get the following error
    00:05:11.952  -- Running vcpkg install
    00:05:11.952  Fetching baseline information from https://github.com/<ORGANIZATION>/custom-registry-repo...
    00:05:12.896  Error: Couldn't find baseline `"e5............................."` for repo https://github.com/<ORGANIZATION>/custom-registry-repo:
    00:05:12.896  fatal: path 'versions/baseline.json' exists on disk, but not in 'e5...........................................'
    00:05:12.896  
    00:05:12.896  Error: Failed to fetch https://github.com/<ORGANIZATION>/custom-registry-repo:
    00:05:12.896  Error: Failed to fetch ref e5.............................. from repository https://github.com/<ORGANIZATION>/custom-registry-repo.
    00:05:12.896  fatal: Unable to persist credentials with the 'wincredman' credential store.

    00:05:12.896  See https://aka.ms/gcmcore-credstores for more information.
    00:05:12.896  bash: /dev/tty: No such device or address
    00:05:12.896  error: failed to execute prompt script (exit code 1)
    00:05:12.896  fatal: could not read Username for 'https://github.com': No such file or directory

The question is: why fetching is stuck and why I am seeing the second error?

**Comment by AbdulAbCellera** [comment, marked as answer]
The issue turned out to be related to:

-  the git creds manager on the slave and 
- the [Jenkins ssh plug-in ](https://plugins.jenkins.io/ssh-agent/). 

GCM, [according to this](https://github.com/GitCredentialManager/git-credential-manager/blob/main/docs/configuration.md#credentialcredentialstore), does not support `credential.helper wincredman` over ssh connections :( 

so for now to solve this

- the helper is set to store option (UNSECURE) or
- set it to dpapi (needs more steps)

**Reply by tarekahf** [reply]
This means, I had to use this command:
```
git config --global credential.helper store
git config --system --unset credential.helper  #to stop the fatal error reported above
```

So that I was able to cache the credentials with the UNSECURE option on a target machine with a similar setup.
